analyzer/src/main/kotlin/managers/Npm.kt

```diff
@@ -252,26 +252,31 @@ open class Npm(
 
                         response.body?.let { body ->
                             val packageInfo = jsonMapper.readTree(body.string())
-
-                            packageInfo["versions"][version]?.let { versionInfo ->
-                                description = versionInfo["description"].textValueOrEmpty()
-                                homepageUrl = versionInfo["homepage"].textValueOrEmpty()
-
-                                versionInfo["dist"]?.let { dist ->
-                                    downloadUrl = dist["tarball"].textValueOrEmpty().let { tarballUrl ->
-                                        if (tarballUrl.startsWith("http://registry.npmjs.org/")) {
-                                            // Work around the issue described at
-                                            // https://npm.community/t/some-packages-have-dist-tarball-as-http-and-not-https/285/19.
-                                            "https://" + tarballUrl.removePrefix("http://")
-                                        } else {
-                                            tarballUrl
+                            if (packageInfo["versions"] != null) {
```

Comment:
When you say in the commit message "does not contain any version entries" I would have expected that `packageInfo["versions"]` is present but empty. However, here you're checking whether `packageInfo["versions"]` is not present at all.

Anyway, the case that `packageInfo["versions"]` is present but empty is already covered by `?.` after `[version]`, so we really only need to check for `packageInfo["versions"]` not being present at all, which we can do without increasing the nesting level by doing

    packageInfo["versions"]?.get(version)?.let { versionInfo ->

Reply:
Also you commit message needs for rework:

- Please do not exceed the title line beyond about 75 chars,
- prefix the title with the "scope" of your changes, which for single-file changes is just the file name (without extension),
- write "nullpointerException" as "NullPointerException".

So how about this:

> Npm: Avoid a NullPointerException if the "versions" node is not present

Reply:
Thanks @sschuberth, agreed on all points. Commit messages amended and code refactored accordingly.

Reply:
Unfortunately, now the Git history is messed up as it contains 4 (!) commits. Please squash them all into one commit (you should have just amended your existing commit and force-pushed to address review comments).